### ISSUE TITLE
BeforeSuite fails in node e2e test

### ISSUE THREAD

**Issue by mwielgus** [kubernetes/kubernetes#30779]
```
BeforeSuite 10s
go run hack/e2e.go -v -test --test_args='--ginkgo.focus=BeforeSuite$'
/var/lib/jenkins/workspace/node-pull-build-e2e-test/go/src/k8s.io/kubernetes/_output/local/go/src/k8s.io/kubernetes/test/e2e_node/e2e_node_suite_test.go:145
Node 1 disappeared before completing BeforeSuite
/var/lib/jenkins/workspace/node-pull-build-e2e-test/go/src/k8s.io/kubernetes/_output/local/go/src/k8s.io/kubernetes/test/e2e_node/e2e_node_suite_test.go:145from junit_gci-dev-54-8711-0-006.xml
Filter through log files
```

**Comment by yujuhong** [comment]
@mwielgus it'd be better to include a gubernator link or a google cloud storage link to the build, so that people can look through the information.

If the log means what it said "Node 1 disappeared before completing BeforeSuite", perhaps the node was deleted while the test was running? 
/cc @kubernetes/goog-node @dchen1107 to see if anyone deleted the VM by accident.

**Comment by Random-Liu** [comment]
Here the "node" is ginkgo parallel test node. https://github.com/onsi/ginkgo/blob/master/integration/suite_setup_test.go#L174

Given the image is gci, I guess this should be https://github.com/kubernetes/kubernetes/issues/29932.
https://github.com/kubernetes/kubernetes/pull/30198 should fix this.

**Comment by Random-Liu** [comment]
I checked the log, and found that the test deletes the instance with no reason, we really need more log...:
```
I0817 08:43:22.148349   22308 e2e_remote.go:112] Building archive...
I0817 08:43:35.985496   22308 e2e_remote.go:164] Staging test binaries on tmp-node-e2e-b253509e-gci-dev-54-8711-0-0
I0817 08:43:35.985770   22308 e2e_remote.go:164] Staging test binaries on tmp-node-e2e-b253509e-coreos-alpha-1122-0-0-v20160727
I0817 08:43:35.987038   22308 e2e_remote.go:164] Staging test binaries on tmp-node-e2e-b253509e-e2e-node-ubuntu-trusty-docker10-v1-image
I0817 08:43:35.987261   22308 e2e_remote.go:164] Staging test binaries on tmp-node-e2e-b253509e-e2e-node-ubuntu-trusty-docker9-v1-image
I0817 08:43:35.988858   22308 e2e_remote.go:164] Staging test binaries on tmp-node-e2e-b253509e-e2e-node-containervm-v20160321-image
I0817 08:43:37.890455   22308 e2e_remote.go:205] Killing any existing node processes on tmp-node-e2e-b253509e-coreos-alpha-1122-0-0-v20160727
I0817 08:43:37.905797   22308 e2e_remote.go:205] Killing any existing node processes on tmp-node-e2e-b253509e-gci-dev-54-8711-0-0
I0817 08:43:38.049219   22308 e2e_remote.go:210] Extracting tar on tmp-node-e2e-b253509e-gci-dev-54-8711-0-0
I0817 08:43:38.096726   22308 e2e_remote.go:210] Extracting tar on tmp-node-e2e-b253509e-coreos-alpha-1122-0-0-v20160727
I0817 08:43:38.370571   22308 e2e_remote.go:205] Killing any existing node processes on tmp-node-e2e-b253509e-e2e-node-ubuntu-trusty-docker10-v1-image
I0817 08:43:38.602389   22308 e2e_remote.go:205] Killing any existing node processes on tmp-node-e2e-b253509e-e2e-node-ubuntu-trusty-docker9-v1-image
I0817 08:43:38.714733   22308 e2e_remote.go:205] Killing any existing node processes on tmp-node-e2e-b253509e-e2e-node-containervm-v20160321-image
I0817 08:43:38.740423   22308 e2e_remote.go:210] Extracting tar on tmp-node-e2e-b253509e-e2e-node-ubuntu-trusty-docker10-v1-image
I0817 08:43:38.814811   22308 e2e_remote.go:210] Extracting tar on tmp-node-e2e-b253509e-e2e-node-containervm-v20160321-image
I0817 08:43:38.982602   22308 e2e_remote.go:210] Extracting tar on tmp-node-e2e-b253509e-e2e-node-ubuntu-trusty-docker9-v1-image
I0817 08:43:41.089352   22308 e2e_remote.go:223] Starting tests on tmp-node-e2e-b253509e-coreos-alpha-1122-0-0-v20160727
I0817 08:43:41.122027   22308 e2e_remote.go:223] Starting tests on tmp-node-e2e-b253509e-gci-dev-54-8711-0-0
I0817 08:43:41.822479   22308 e2e_remote.go:223] Starting tests on tmp-node-e2e-b253509e-e2e-node-ubuntu-trusty-docker10-v1-image
I0817 08:43:42.006772   22308 e2e_remote.go:223] Starting tests on tmp-node-e2e-b253509e-e2e-node-containervm-v20160321-image
I0817 08:43:42.400886   22308 e2e_remote.go:223] Starting tests on tmp-node-e2e-b253509e-e2e-node-ubuntu-trusty-docker9-v1-image
I0817 08:43:54.140162   22308 e2e_remote.go:230] Copying test artifacts from tmp-node-e2e-b253509e-gci-dev-54-8711-0-0
I0817 08:43:54.408724   22308 run_e2e.go:506] Deleting instance "tmp-node-e2e-b253509e-gci-dev-54-8711-0-0"
================================================================
Failure Finished Host tmp-node-e2e-b253509e-gci-dev-54-8711-0-0 Test Suite
Running Suite: E2eNode Suite
============================
Random Seed: 1471448621 - Will randomize all specs
Will run 90 of 103 specs

Running in parallel across 8 nodes

Failure [10.470 seconds]
[BeforeSuite] BeforeSuite 
/var/lib/jenkins/workspace/node-pull-build-e2e-test/go/src/k8s.io/kubernetes/_output/local/go/src/k8s.io/kubernetes/test/e2e_node/e2e_node_suite_test.go:145

  Node 1 disappeared before completing BeforeSuite

  /var/lib/jenkins/workspace/node-pull-build-e2e-test/go/src/k8s.io/kubernetes/_output/local/go/src/k8s.io/kubernetes/test/e2e_node/e2e_node_suite_test.go:145
```

And then the test fail.

https://console.cloud.google.com/storage/browser/kubernetes-jenkins/pr-logs/pull/30207/node-pull-build-e2e-test/19417

**Comment by Random-Liu** [comment]
https://k8s-gubernator.appspot.com/build/kubernetes-jenkins/pr-logs/pull/30623/node-pull-build-e2e-test/19432/

**Comment by Random-Liu** [comment]
This is the same issue with #29932.

After https://github.com/kubernetes/kubernetes/pull/30116, the log of "e2e services" will be printed into `services.log`. And in the log, we can see:
```
F0817 17:21:24.481519    1415 e2e_service.go:116] Failed to run e2e services: etcd binary not found
goroutine 35 [running]:
k8s.io/kubernetes/vendor/github.com/golang/glog.stacks(0x4456900, 0x0, 0x0, 0x0)
	/var/lib/jenkins/workspace/node-pull-build-e2e-test/go/src/k8s.io/kubernetes/_output/local/go/src/k8s.io/kubernetes/vendor/github.com/golang/glog/glog.go:766 +0xb8
k8s.io/kubernetes/vendor/github.com/golang/glog.(*loggingT).output(0x4436020, 0xc800000003, 0xc82000c6c0, 0x4120a88, 0xe, 0x74, 0x0)
	/var/lib/jenkins/workspace/node-pull-build-e2e-test/go/src/k8s.io/kubernetes/_output/local/go/src/k8s.io/kubernetes/vendor/github.com/golang/glog/glog.go:717 +0x259
k8s.io/kubernetes/vendor/github.com/golang/glog.(*loggingT).printf(0x4436020, 0x7f0600000003, 0x306ee00, 0x1e, 0xc820aedce0, 0x1, 0x1)
	/var/lib/jenkins/workspace/node-pull-build-e2e-test/go/src/k8s.io/kubernetes/_output/local/go/src/k8s.io/kubernetes/vendor/github.com/golang/glog/glog.go:655 +0x1d4
k8s.io/kubernetes/vendor/github.com/golang/glog.Fatalf(0x306ee00, 0x1e, 0xc820aedce0, 0x1, 0x1)
	/var/lib/jenkins/workspace/node-pull-build-e2e-test/go/src/k8s.io/kubernetes/_output/local/go/src/k8s.io/kubernetes/vendor/github.com/golang/glog/glog.go:1145 +0x5d
k8s.io/kubernetes/test/e2e_node.RunE2EServices()
	/var/lib/jenkins/workspace/node-pull-build-e2e-test/go/src/k8s.io/kubernetes/_output/local/go/src/k8s.io/kubernetes/test/e2e_node/e2e_service.go:116 +0x5a6
k8s.io/kubernetes/test/e2e_node.TestE2eNode(0xc82007aab0)
	/var/lib/jenkins/workspace/node-pull-build-e2e-test/go/src/k8s.io/kubernetes/_output/local/go/src/k8s.io/kubernetes/test/e2e_node/e2e_node_suite_test.go:68 +0x66
testing.tRunner(0xc82007aab0, 0x4329750)
	/usr/local/go/src/testing/testing.go:473 +0x98
created by testing.RunTests
	/usr/local/go/src/testing/testing.go:582 +0x892
```

Then, the test failed and we "copy test artifacts" and "delete the instance".

**Comment by yujuhong** [comment]
Sounds like the cleanup order is wrong if the suite deleted the VM instance before ginkgo terminated properly.

**Comment by Random-Liu** [comment]
It is because we run the test remotely, and print test result after the test finished. https://github.com/kubernetes/kubernetes/blob/master/test/e2e_node/runner/run_e2e.go#L234-L244

At that time, the cleanup has already been done. :)

**Comment by yujuhong** [comment]
@Random-Liu, the problem is that this error message is masking the real errors. We should try to fix this if we can...

**Comment by Random-Liu** [comment]
This should have been fixed by #30198.
And the comment is addressed by #30941.
Close this one.

Feel free to reopen if it happens again. :)
